subsys/bluetooth/mesh/gen_loc_srv.c

```diff
@@ -194,37 +185,60 @@ const struct bt_mesh_model_op _bt_mesh_loc_setup_srv_op[] = {
 	BT_MESH_MODEL_OP_END
 };
 
-static int update_handler(struct bt_mesh_model *model)
+static void global_update_handler(struct bt_mesh_loc_srv *srv)
 {
-	struct bt_mesh_loc_srv *srv = model->user_data;
+	struct bt_mesh_loc_global loc = LOC_GLOBAL_DEFAULT;
 
-	if (srv->pub_op == BT_MESH_LOC_OP_GLOBAL_STATUS) {
-		struct bt_mesh_loc_global loc = LOC_GLOBAL_DEFAULT;
+	srv->pub_state.last = 0;
+	srv->handlers->global_get(srv, NULL, &loc);
 
-		srv->handlers->global_get(srv, NULL, &loc);
+	bt_mesh_model_msg_init(srv->pub.msg, BT_MESH_LOC_OP_GLOBAL_STATUS);
+	bt_mesh_loc_global_encode(srv->pub.msg, &loc);
+}
 
-		bt_mesh_model_msg_init(srv->pub.msg,
-				       BT_MESH_LOC_OP_GLOBAL_STATUS);
-		bt_mesh_loc_global_encode(srv->pub.msg, &loc);
-	} else if (srv->pub_op == BT_MESH_LOC_OP_LOCAL_STATUS) {
-		struct bt_mesh_loc_local loc = LOC_LOCAL_DEFAULT;
+static void local_update_handler(struct bt_mesh_loc_srv *srv)
+{
+	struct bt_mesh_loc_local loc = LOC_LOCAL_DEFAULT;
 
-		srv->handlers->local_get(srv, NULL, &loc);
+	srv->pub_state.last = 1;
```

Comment:
I'm not a fan of these magic numbers `1` and `0`. Either create #defines or an enum for them, or make the field a bool called something like `last_was_local`.

Reply:
The other fields should also be `bool`, and renamed `local_available` and `global_availble`. Makes it easier to follow when reading the code without having the struct with doc comments available on the same screen.

Reply:
I do not see the reason to have `bool` for them and spend 3(probably more if considering potential structure alignment) bytes if it is possible to use bitfields.

Reply:
At least rename the fields with `_available`. It is not obvious when reading the code without having the struct definition at hand.